docs/_docs/01_getting-started/terminology.md

```diff
@@ -0,0 +1,219 @@
+
+Units typically represent a minimal useful piece of infrastructure that should be independently managed.
+
+e.g. A unit might represent a single VPC, a single database, or a single server.
+
+While not a requirement, a general tendency experienced when working with Terragrunt is that units tend to decrease in size. This is because Terragrunt makes it easy to segment pieces of infrastructure into their own state, and to have them interact with each other through the use of [dependency blocks](/docs/reference/config-blocks-and-attributes#dependency). Smaller units are quicker to update, easier to reason about and safer to work with.
+
+A common pattern used in the repository structure for Terragrunt projects is to have a single `terragrunt.hcl` file located at the root of the repository, and multiple subdirectories each containing their own `terragrunt.hcl` file. This is typically done to promote code-reuse, as it allows for any configuration common to all units to be defined in the root `terragrunt.hcl` file, and for unit-specific configuration to be defined in child directories. In this pattern, the root `terragrunt.hcl` file is not considered a unit, while all the child directories containing `terragrunt.hcl` files are.
+
+Note that units don't technically need to call their configuration files `terragrunt.hcl` (that's configurable via the [--terragrunt-config](/docs/reference/cli-options/#terragrunt-config)), and users don't technically need to use a root `terragrunt.hcl` file or to name it that. This is the most common pattern followed by the community, however, and deviation from this pattern should be justified in the context of the project. It can help others with Terragrunt experience understand the project more easily if industry standard patterns are followed.
+
+### Stack
+
+A stack is a collection of units managed by Terragrunt. There is ([as of writing](https://github.com/gruntwork-io/terragrunt/issues/3313) work underway to provide a top level artifact for interacting with stacks via a `terragrunt.stack.hcl` file, but as of now, stacks are generally defined by a directory with a tree of units. Units within a stack can be dependent on each other, and can be updated in a specific order to ensure that dependencies are resolved in the correct order.
+
+Stacks typically represent a collection of units that need to be managed in concert.
+
+e.g. A stack might represent a collection of units that together form a single application environment, a business unit, or a region.
+
+The design of `terragrunt.stack.hcl` files is to ensure that they function entirely as a convenient shorthand for an equivalent directory structure of units. This is to ensure that users are able to easily transition between the two paradigms, and are able to decide for themselves which approach to structuring infrastructure is most appropriate for their use case.
+
+### Module
```

Comment:
Should we define modules after OpenTofu and before units? This way, we can explain units as an instance of a module.

Reply:
I can be persuaded, but my thought process is that we should define Terragrunt specific terms before OpenTofu terms.

Note that I didn't reference modules in the definition of units. Technically speaking, you don't _have_ to use modules with units. You can just write `.tf` files adjacent to `terragrunt.hcl` files. It's just a best practice to define all infrastructure in modules.